.github/workflows/csound_builds.yml

```diff
@@ -124,6 +124,119 @@ jobs:
       - name: Run integration and unit tests
         run: cmake --build build --target test csdtests
 
+  macos_build_osxcross:
+    name: MacOS build (osxcross)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+          submodules: true
+
+      - name: Bootstrap VCPKG
+        run: ./vcpkg/bootstrap-vcpkg.sh
+
+      - name: Setup NuGet credentials
+        shell: bash
+        run: |
+          mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+          sources add \
+          -source "https://nuget.pkg.github.com/csound/index.json" \
+          -storepasswordincleartext \
+          -name "GitHub" \
+          -username "csound" \
+          -password "${{ secrets.GITHUB_TOKEN }}"
+          mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+            setapikey "${{ secrets.GITHUB_TOKEN }}" \
+            -source "https://nuget.pkg.github.com/csound/index.json"
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install build-essential liblzma-dev libxml2-dev libz-dev libbz2-dev cpio ninja-build
+
+      - name: Cache osxcross
+        id: osxcross
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/work/csound/csound/osxcross
+          key: ${{ runner.os }}-osxcross
+  
+      - name: Build osxcross
+        if: steps.osxcross.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/tpoechtrager/osxcross.git
+          sudo /home/runner/work/csound/csound/osxcross/tools/get_dependencies.sh
+          OSX_SDK="MacOSX14.0.sdk"
+          OSX_SDK_URL="https://github.com/joseluisq/macosx-sdks/releases/download/14.0/${OSX_SDK}.tar.xz"
+          curl -sSL "$OSX_SDK_URL" -o "/home/runner/work/csound/csound/osxcross/tarballs/$OSX_SDK.tar.xz"
+          UNATTENDED=1 /home/runner/work/csound/csound/osxcross/build_clang.sh
+          UNATTENDED=1 /home/runner/work/csound/csound/osxcross/build.sh
```

Comment:
How big is this to cache do you know?

Reply:
The cache is for this is 1.7 GB

Reply:
That might be too much. There is already the cache for each VCPKG dir and we have to share that with every platform. Also I didn't realise this takes 2.5 hrs to build. Is there anyway to run this locally and cache this in a URL elsewhere?

Reply:
Yeah, it should be possible to cache this somewhere.  Is there some location that would be preferred?

Reply:
Hmm not really sure. Somewhere publicly hosted I guess